### ISSUE TITLE
Recurrent Dropbox "conflicted copy" on Preferences for: SizeUp, Bartender, Sketch

### ISSUE THREAD

**Issue by jm3** [lra/mackup#1312]
<img width="869" alt="conflicted-copies" src="https://user-images.githubusercontent.com/12213/51938733-b04d6300-23c2-11e9-90a2-d22bbfe616c0.png">

Is it normal that preferences for supported apps like SizeUp, Bartender, and Sketch keep creating Dropbox conflicts every day like this? After the initial setup, I haven't subsequently touched Mackup to sync or restore — this appears to be a pure App / Dropbox interaction situation. I'm signed into the same Dropbox account on all boxes.

Are these constant conflicts (and accompanying Dropbox pop-up notificatons) something that Mackup users just live with? I can stop syncing those apps, but I'm confused as to why this even happens. I assume these apps are writing to their preferences in such a way that misleads Dropbox to think there are conflicts, so should they still be listed as Mackup-supported?

**Comment by lra** [comment]
Not sure what's triggering this as I don't use those apps.
Might happen if those apps change the file in the same time frame, e.g. you change the file on A, A being offline, and you change the file on B, B syncs up, and A goes online later, resulting on a conflict.

Do you you have a workflow like this?

**Comment by jm3** [comment]
Nope, these are machines that aren't used at the same time, i.e. one is home, one is at work.

**Comment by lra** [comment]
This is more a dropbox problem than a mackup one, those type of files are created by dropbox when it can't figure out which of the conflicting versions of a file is the legit one, as the same file has been changed in parallel during the same timeframe.

I don't think it's due to the applications themselves, but to a specific workflow.

**Comment by jm3** [comment]
I understand what you mean. There's no real "workflow" other than the app is running. I don't touch the app, the prefs files, or even reboot. It seems like, when I awaken the mac, and the apps are already running, as is Dropbox, so even though the files are changed hours or even several days apart, with no risk of a sync conflict, Dropbox tries to bring the box up to date at the same time that some of these apps want to aggressively flush their preferences to disk, causing the sync collision / conflict.

What I'm trying to be clear about is: I'm not using these apps in any unusual way, and I'm not using them at the same time, and the files are not being changed at the same time. It seems to be just that these apps are very eager to re-save their prefs, especially upon wakeup? Since there's no workflow here, and no unusual usage, and this is the normal use case for Mackup (at least as far as I understand), should these apps be marked as unsupported if their default behavior is incompatible with Dropbox and/or Mackup?

Otherwise I assume I should be exploring alternate backing stores for Mackup than Dropbox. I'm open to using e.g. iCloud or something else, but in my experience, Dropbox has been the best at syncing changes quickly with the fewest conflicts. I've used it for a decade without issues for large volumes accessed by many users. That's what I'm scratching my head at — this problem would seem to happen for everyone using Mackup with Dropbox with a preference file that frequently updates itself, right?

**Comment by bensurtees** [comment]
Hi, I am the creator of Bartender, happy to help with this if I can, not sure what I can add. jm3 wanted me to explain how Bartender writes its preferences, and this may be useful generally, as Bartender does nothing special with its preferences at all. 
All Bartender preferences are read/written by Bartender using the UserDefaults system in macOS. As is now recommended, we never synchronise or flush these to disc/file, and leave this to the defaults system in macOS to do when it deems appropriate (often when the app quits and other more random times).

Ben

**Comment by lra** [comment]
> I completely understand that this is not a Mackup problem. My confusion is more the idea that these apps are marked "supported", but don't seem to work reliably when using Dropbox.

You are totally right, and if it's the case, we can remove the apps from the official list, which still allow people to add the apps back themselves.

I just want to be sure it's a app-related problem first, and not related to your setup and workflow.

**Comment by jm3** [comment]
@bensurtees: got it, thank you for clarifying!

@lra: I'm far from knowledgeable about this, but could [SizeUp](http://www.irradiatedsoftware.com/sizeup/) and [Sketch](https://www.sketchapp.com/) also be following those same macOS UserDefaults guidelines like Bartender?, E.g., in this list of recently "deleted" Dropbox files, which I didn't delete — they just delete and re-appear continuously without me touching any preference settings at all, you can see that they all use the same trailing naming convention.

<img width="987" alt="dropbox sync conflicts on prefs" src="https://user-images.githubusercontent.com/12213/52009151-18fe1380-2487-11e9-9f6e-33a4f2d87669.png">

**Comment by jm3** [comment]
@bensurtees: got it, that makes sense.

@lra: does this mean that os x apps using apple's guidelines for saving preferences should not be managed by mackup?

**Comment by lra** [comment]
> @lra: does this mean that os x apps using apple's guidelines for saving preferences should not be managed by mackup?

No clue, I don't know much about those guidelines. Those temp files should no be a problem at all, those dropbox conflicts are though. We should try to understand what triggers them.

**Comment by jm3** [comment]
Looks like Spotify now follows those OS X guidelines as well:

<img width="432" alt="spotify" src="https://user-images.githubusercontent.com/12213/52185246-35c18080-27d2-11e9-9927-01e3f2741e29.png">

**Comment by jm3** [comment]
…as does OS X Terminal.

**Comment by jasongregori** [comment]
I have the same issues. There seems to be something about if the app was already running before I did the Mackup restore. I just set up 2 new computers. On one I have the constant dropbox conflicts and alerts for both Xcode, bartender, and moom. On the second computer which I was more careful about (I quit all the apps, did the Mackup restore, then killed cfprefsd) it *seems* like I'm not getting the alerts for Xcode and moom. However I then realized I actually forgot to quit bartender and I'm still getting the alerts for bartender... 

It almost seems like bartender maybe got some state before I restored and then keeps trying to sync that state to user defaults and it keeps getting overwritten by bartender on the original machine. (Note I have my older machine and one of the new ones running at the same time).

No idea if that's helpful but thought I'd chime in.
